code/modules/admin/banjob.dm

```diff
@@ -1,109 +1,21 @@
-//This file was auto-corrected by findeclaration.exe on 25.5.2012 20:42:32
-
-var/jobban_runonce			// Updates legacy bans with new info
-var/jobban_keylist[0]		//to store the keys & ranks
-
-/proc/jobban_fullban(mob/M, rank, reason)
-	if (!M || !M.key) return
-	jobban_keylist.Add(text("[M.ckey] - [rank] ## [reason]"))
-	jobban_savebanfile()
-
-/proc/jobban_client_fullban(ckey, rank)
-	if (!ckey || !rank) return
-	jobban_keylist.Add(text("[ckey] - [rank]"))
-	jobban_savebanfile()
-
 //returns a reason if M is banned from rank, returns 0 otherwise
 /proc/jobban_isbanned(mob/M, rank)
-	if(M && rank)
-		for (var/s in jobban_keylist)
-			if( findtext(s,"[M.ckey] - [rank]") == 1 )
-				var/startpos = findtext(s, "## ")+3
-				if(startpos && startpos<length(s))
-					var/text = copytext(s, startpos, 0)
-					if(text)
-						return text
-				return "Reason Unspecified"
+	if(!M || !M.client)

```

Comment:
if the mob has a ckey or key, but no client. you should make this fall back to a db query